.buildkite/steps/packer.sh

```diff
@@ -16,17 +16,12 @@ fi
 
 mkdir -p "build/"
 
-if [[ "$os" == "linux" ]]; then
-  buildkite-agent artifact download "build/fix-perms-linux-${arch}" ./build
-  mv "build/fix-perms-linux-${arch}" packer/linux/conf/buildkite-agent/scripts/fix-buildkite-agent-builds-permissions
-  chmod 755 packer/linux/conf/buildkite-agent/scripts/fix-buildkite-agent-builds-permissions
-fi
-
 # Build a hash of packer files and the agent versions
 packer_files_sha=$(find Makefile "packer/${os}" plugins/ -type f -print0 | xargs -0 sha1sum | awk '{print $1}' | sort | sha1sum | awk '{print $1}')
+internal_files_sha=$(find go.mod go.sum internal/ -type f -print0 | xargs -0 sha1sum | awk '{print $1}' | sort | sha1sum | awk '{print $1}')
 stable_agent_sha=$(curl -Lfs "https://download.buildkite.com/agent/stable/latest/${agent_binary}.sha256")
 unstable_agent_sha=$(curl -Lfs "https://download.buildkite.com/agent/unstable/latest/${agent_binary}.sha256")
-packer_hash=$(echo "$packer_files_sha" "$arch" "$stable_agent_sha" "$unstable_agent_sha" | sha1sum | awk '{print $1}')
+packer_hash=$(echo "$packer_files_sha" "$internal_files_sha" "$arch" "$stable_agent_sha" "$unstable_agent_sha" | sha1sum | awk '{print $1}')
```

Comment:
Maybe we shouldn't use sha1 in 2023. I guess someone with keys to our account could pre-build an malicious AMI, upload its sha1 to the s3 bucket and submit an innocuous PR that collides with the malicious AMI. But if they had the keys to build an AMI, they could just replace an existing released AMI with the malicious one. So I don't think there's a need to change this.

Reply:
Changing the inputs to the hash seems like a good time to change the hash function.